cdk/kittyhawk/src/service.ts

```diff
@@ -2,40 +2,22 @@ import { Construct } from 'constructs';
 import { KubeService as ServiceApiObject, IntOrString } from './imports/k8s';
 
 
-export interface ServiceProps {
-  /**
-   * External port.
-   *
-   * @default 80
-   */
-  readonly port?: number;
-
-  /**
-* Internal port.
-*
-* @default port
-*/
-  readonly containerPort?: number;
-}
-
-
 export class Service extends Construct {
-  constructor(scope: Construct, appname: string, props: ServiceProps) {
+  constructor(scope: Construct, appname: string, port?: number) {
     super(scope, `service-${appname}`);
 
-    const port = props.port || 80;
-    const containerPort = props.containerPort || port;
+    const targetPort = port ?? 80;
 
     new ServiceApiObject(this, `service-${appname}`, {
```

Comment:
You're talking about constructs from `src/imports` right (like `ServiceApiObject`, `CronjobApiObject`, `IngressApiObject`). 

What's the use case that developers who are using kittyhawk to deploy can call `ApiObject.of` of an application or cronjob construct? Or do you mean using `ApiObject.of` within the kittyhawk code itself in the future?

Either way I don't see any harm in doing this for all of the ApiObjects so far and it could be helpful. Do you think they will change the name from `Default` to something else in the future?

Some lower level objects (L2?) seem to be called `Resource` now: [link](https://github.com/cdk8s-team/cdk8s/commit/a8337e805dce81e0d0883bb296c1017b5ac2311f)

Reply:
Actually wait, or do you mean not just the import Api objects, but ALL children Api objects in general?

Reply:
I think naming the children Api objects `Default` make sense since we're not really using the object names for anything. We should make sure that's something that if it might get changed later, we can quickly change it in one place.

For example, in utils.ts have `defaultChildName = "Default"`

Reply:
Okay updated children name in this [commit](https://github.com/pennlabs/infrastructure/pull/40/commits/6bf82b9e1a572208311993b23fd5183d6121d857). I think doing it this way is good. Feel free to take a look